File: lib/rgfa/line/group/ordered/to_gfa1.rb
module RGFA::Line::Group::Ordered::ToGFA1

  def to_gfa1_a
    a = ["P"]
    if name.placeholder?
      raise RGFA::ValueError,
        "Conversion to GFA1 failed\n"+
        "The path name is a placeholder\t"+
        "Line: #{self}"
    end
    a << name.to_s
    segment_names = []
    captured_segments.each do |oline|
      oline.name.validate_gfa_field(:segment_name_gfa1)
      segment_names << oline.to_s
    end
    a << segment_names.join(",")
    overlaps = []
    captured_edges.each do |oline|
      oline.line.overlap.validate_gfa_field(:alignment_gfa1)
      overlaps << oline.line.overlap.to_s
    end
    a << overlaps.join(",")
    return a
  end

end

File: gfapy/line/group/ordered/to_gfa1.py
import gfapy

class ToGFA1:

  def _to_gfa1_a(self):
    a = ["P"]
    if gfapy.is_placeholder(self.name):
      raise gfapy.ValueError(
        "Conversion to GFA1 failed\n"+
        "The path name is a placeholder\t"+
        "Line: {}".format(self))
    a.append(self.name)
    segment_names = []
    for oline in self.captured_segments:
      gfapy.Field._validate_gfa_field(oline.name, "segment_name_gfa1")
      segment_names.append(str(oline))
    a.append(",".join(segment_names))
    overlaps = []
    for oline in self.captured_edges:
      gfapy.Field._validate_gfa_field(oline.line.overlap, "alignment_gfa1")
      overlaps.append(str(oline.line.overlap))
    a.append(",".join(overlaps))
    for tn in self.tagnames:
      a.append(self.field_to_s(tn, tag=True))
    return a
